frontend/src/metabase/dashboard/actions/getNewCardUrl.ts

```diff
@@ -0,0 +1,103 @@
+import _ from "underscore";
+
+import { getCardAfterVisualizationClick } from "metabase/visualizations/lib/utils";
+import * as Lib from "metabase-lib";
+import Question from "metabase-lib/v1/Question";
+import type Metadata from "metabase-lib/v1/metadata/Metadata";
+import type { ParameterWithTarget } from "metabase-lib/v1/parameters/types";
+import * as ML_Urls from "metabase-lib/v1/urls";
+import type {
+  Card,
+  Dashboard,
+  QuestionDashboardCard,
+} from "metabase-types/api";
+import type { State } from "metabase-types/store";
+
+export const getNewCardUrl = (
```

Comment:
This function has been extracted from `navigateToNewCardFromDashboard` with only minor changes